operator_v2/validator/controller.go

```diff
@@ -207,26 +207,26 @@ func (c *controller) setupNetworkHandlers() error {
 	return nil
 }
 
-func (c *controller) GetAllValidatorShares() ([]*beaconprotocol.Share, error) {
+func (c *controller) GetAllValidatorShares() ([]*share.Share, error) {
 	return c.collection.GetAllValidatorShares()
 }
 
 func (c *controller) GetValidatorStats() (uint64, uint64, uint64, error) {
-	allShares, err := c.collection.GetAllValidatorShares()
+	allMetadata, err := c.collection.GetAllValidatorMetadata()
```

Comment:
Let's use 1 function for saving and fetch. also lets wrap both share and metadata under one struct